go/enclave/enclave.go

```diff
@@ -87,6 +87,7 @@ type enclaveImpl struct {
 
 	enclaveKey    *ecdsa.PrivateKey // this is a key specific to this enclave, which is included in the Attestation. Used for signing rollups and for encryption of the shared secret.
```

Comment:
feels like we should group these fields and the logic into a struct that only has that responsibility.

Reply:
Yeah good shout. Is the new commit the sort of thing you had in mind?